openslides_backend/presenter/check_mediafile_id.py

```diff
@@ -37,7 +48,99 @@ def get_result(self) -> Any:
+                "projection_ids",
+                "is_public",
+                "inherited_access_group_ids",
+            ],
+        )
+        meeting_id = mediafile.get("meeting_id")
+        if not meeting_id:
+            raise PermissionDenied("You are not allowed to see this mediafile.")
+
+        meeting = self.datastore.get(
+            FullQualifiedId(Collection("meeting"), meeting_id),
+            ["admin_group_id", "enable_anonymous", "user_ids", "committee_id"],
+        )
+        groups_field = f"group_${meeting_id}_ids"
+        user = self.datastore.get(
+            FullQualifiedId(Collection("user"), self.user_id), [groups_field]
+        )
+        if meeting.get("admin_group_id") in user.get(groups_field, []):
+            return
+        # The user can see the meeting and (used_as_logo_$_in_meeting_id
+        #    or used_as_font_$_in_meeting_id is not empty)
+        can_see_meeting = self.check_can_see_meeting(meeting)
+        if can_see_meeting:
+            if mediafile.get("used_as_logo_$_in_meeting_id") or mediafile.get(
+                "used_as_font_$_in_meeting_id"
+            ):
+                return
+        # The user has projector.can_see
+        # and there exists a mediafile/projection_ids with
+        # projection/current_projector_id set
+        if has_perm(
+            self.datastore, self.user_id, Permissions.Projector.CAN_SEE, meeting_id
+        ):
+            for projection_id in mediafile.get("projection_ids", []):
+                projection = self.datastore.get(
+                    FullQualifiedId(Collection("projection"), projection_id),
+                    ["current_projector_id"],
+                )
+                if projection.get("current_projector_id"):
+                    return
+        # The user has mediafile.can_see and either:
+        #  - mediafile/is_public is true, or
+        #   - The user has groups in common with mediafile/inherited_access_group_ids
+        if has_perm(
+            self.datastore, self.user_id, Permissions.Mediafile.CAN_SEE, meeting_id
+        ):
+            if mediafile.get("is_public"):
+                return
+            inherited_access_group_ids = set(
+                mediafile.get("inherited_access_group_ids", [])
+            )
+            user = self.datastore.get(
+                FullQualifiedId(Collection("user"), self.user_id),
+                [f"group_${meeting_id}_ids"],
+            )
+            user_groups = set(user.get(f"group_${meeting_id}_ids", []))
+            if inherited_access_group_ids & user_groups:
+                return
+        raise PermissionDenied("You are not allowed to see this mediafile.")
+
+    def check_can_see_meeting(self, meeting: Dict[str, Any]) -> bool:
+        """needs meeting to include enable_anonymous, user_ids, committee_id."""
+        if has_organization_management_level(
+            self.datastore, self.user_id, OrganizationManagementLevel.SUPERADMIN
+        ):
```

Comment:
You can remove this check here since it was already checked earlier in the other function

Reply:
Removed.